protocols/netty/src/main/java/org/apache/james/protocols/netty/ProtocolMDCContextFactory.java

```diff
@@ -96,7 +96,6 @@ static MDCBuilder from(Object o) {
     static MDCBuilder forSession(ProtocolSession protocolSession) {
         return MDCBuilder.create()
             .addToContext(MDCBuilder.SESSION_ID, protocolSession.getSessionID())
-            .addToContext(MDCBuilder.CHARSET, protocolSession.getCharset().displayName())
```

Comment:
@chibenwa @Arsnael I want to point this out and make sure you have seen it and agree with it. 

If I keep it, I end up with the charset mdc attribute being set twice in a single log, unfortunately this makes both logback-classic and logback-contrib json serializer fail to output the log. instead it outputs a message on STDERR 

I have not been able to reproduce this locally in a test, I tried various things such as changing the builder to make sure to clear the MDC and deduplicate the keys manually before writing them back: without success. 
removing this line systematically fixes the problem.

Reply:
Seems legit to me.

The charset info seems to be kinda low value...